docs/auditors-guide.md

```diff
@@ -0,0 +1,113 @@
+# Auditors Guide for Citrea
```

Comment:
Is it an auditors guide (the `docs/` book) or is it a technical documentation which indeed can be used as an auditors guide?